File: test-service-java/src/main/java/com/example/testservicejava/v1/services/UserService.java
package com.example.testservicejava.v1.services;

import com.example.testservicejava.v1.entities.User;
import com.example.testservicejava.v1.repositories.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<User> readAllUsers() {
        return userRepository.findAll();
    }
}

File: test-service-kotlin/src/main/kotlin/org/xapps/service/testservice/v1/services/UserService.kt
package org.xapps.service.testservice.v1.services

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.xapps.service.testservice.v1.entities.User
import org.xapps.service.testservice.v1.repositories.UserRepository

@Service
class UserService @Autowired constructor(
    private val userRepository: UserRepository
) {

    fun readAllUsers(): List<User> =
        userRepository.findAll()

}